core/unit_test/TestComplex.hpp

```diff
@@ -361,6 +361,7 @@ TEST(TEST_CATEGORY, complex_trivially_copyable) {
   // Kokkos::complex<RealType> is trivially copyable when RealType is trivially
   // copyable
 
+#if !defined(__ibmxl__)
 #if !defined(__clang__)
```

Comment:
please do `&&` instead of nested if.

Reply:
These are doing different things. We still check with `clang` assuming that it has `std::is_trivially_copyable`, but we disable the check for `XL`.

Reply:
I don't see how having two separate directives makes that distinction.  Go ahead and add comments though.